Index: examples/Makefile.am
===================================================================
--- examples/Makefile.am	(revision 148515)
+++ examples/Makefile.am	(working copy)
@@ -1,6 +1,6 @@
 bin_PROGRAMS=mtp-connect mtp-detect mtp-tracks mtp-files \
-	mtp-folders mtp-trexist mtp-playlists mtp-getplaylist \
-	mtp-format mtp-albumart mtp-albums mtp-newplaylist mtp-emptyfolders \
+	mtp-folders mtp-trexist \
+	mtp-format mtp-albumart mtp-albums mtp-emptyfolders \
 	mtp-thumb mtp-reset mtp-filetree
 
 mtp_connect_SOURCES=connect.c connect.h delfile.c getfile.c newfolder.c \
@@ -11,9 +11,6 @@
 mtp_files_SOURCES=files.c util.c util.h common.h
 mtp_folders_SOURCES=folders.c util.c util.h common.h
 mtp_trexist_SOURCES=trexist.c util.c util.h common.h
-mtp_playlists_SOURCES=playlists.c util.c util.h common.h
-mtp_getplaylist_SOURCES=getplaylist.c util.c util.h common.h
-mtp_newplaylist_SOURCES=newplaylist.c util.c util.h common.h
 mtp_format_SOURCES=format.c util.c util.h common.h
 mtp_albumart_SOURCES=albumart.c util.c util.h common.h
 mtp_albums_SOURCES=albums.c util.c util.h common.h
Index: examples/Makefile.in
===================================================================
--- examples/Makefile.in	(revision 148515)
+++ examples/Makefile.in	(working copy)
@@ -36,10 +36,10 @@
 host_triplet = @host@
 bin_PROGRAMS = mtp-connect$(EXEEXT) mtp-detect$(EXEEXT) \
 	mtp-tracks$(EXEEXT) mtp-files$(EXEEXT) mtp-folders$(EXEEXT) \
-	mtp-trexist$(EXEEXT) mtp-playlists$(EXEEXT) \
-	mtp-getplaylist$(EXEEXT) mtp-format$(EXEEXT) \
+	mtp-trexist$(EXEEXT) \
+	mtp-format$(EXEEXT) \
 	mtp-albumart$(EXEEXT) mtp-albums$(EXEEXT) \
-	mtp-newplaylist$(EXEEXT) mtp-emptyfolders$(EXEEXT) \
+	mtp-emptyfolders$(EXEEXT) \
 	mtp-thumb$(EXEEXT) mtp-reset$(EXEEXT) mtp-filetree$(EXEEXT)
 subdir = examples
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
@@ -95,18 +95,6 @@
 mtp_format_OBJECTS = $(am_mtp_format_OBJECTS)
 mtp_format_LDADD = $(LDADD)
 mtp_format_DEPENDENCIES = ../src/libmtp.la
-am_mtp_getplaylist_OBJECTS = getplaylist.$(OBJEXT) util.$(OBJEXT)
-mtp_getplaylist_OBJECTS = $(am_mtp_getplaylist_OBJECTS)
-mtp_getplaylist_LDADD = $(LDADD)
-mtp_getplaylist_DEPENDENCIES = ../src/libmtp.la
-am_mtp_newplaylist_OBJECTS = newplaylist.$(OBJEXT) util.$(OBJEXT)
-mtp_newplaylist_OBJECTS = $(am_mtp_newplaylist_OBJECTS)
-mtp_newplaylist_LDADD = $(LDADD)
-mtp_newplaylist_DEPENDENCIES = ../src/libmtp.la
-am_mtp_playlists_OBJECTS = playlists.$(OBJEXT) util.$(OBJEXT)
-mtp_playlists_OBJECTS = $(am_mtp_playlists_OBJECTS)
-mtp_playlists_LDADD = $(LDADD)
-mtp_playlists_DEPENDENCIES = ../src/libmtp.la
 am_mtp_reset_OBJECTS = reset.$(OBJEXT) util.$(OBJEXT)
 mtp_reset_OBJECTS = $(am_mtp_reset_OBJECTS)
 mtp_reset_LDADD = $(LDADD)
@@ -140,16 +128,14 @@
 	$(mtp_connect_SOURCES) $(mtp_detect_SOURCES) \
 	$(mtp_emptyfolders_SOURCES) $(mtp_files_SOURCES) \
 	$(mtp_filetree_SOURCES) $(mtp_folders_SOURCES) \
-	$(mtp_format_SOURCES) $(mtp_getplaylist_SOURCES) \
-	$(mtp_newplaylist_SOURCES) $(mtp_playlists_SOURCES) \
+	$(mtp_format_SOURCES) \
 	$(mtp_reset_SOURCES) $(mtp_thumb_SOURCES) \
 	$(mtp_tracks_SOURCES) $(mtp_trexist_SOURCES)
 DIST_SOURCES = $(mtp_albumart_SOURCES) $(mtp_albums_SOURCES) \
 	$(mtp_connect_SOURCES) $(mtp_detect_SOURCES) \
 	$(mtp_emptyfolders_SOURCES) $(mtp_files_SOURCES) \
 	$(mtp_filetree_SOURCES) $(mtp_folders_SOURCES) \
-	$(mtp_format_SOURCES) $(mtp_getplaylist_SOURCES) \
-	$(mtp_newplaylist_SOURCES) $(mtp_playlists_SOURCES) \
+	$(mtp_format_SOURCES) \
 	$(mtp_reset_SOURCES) $(mtp_thumb_SOURCES) \
 	$(mtp_tracks_SOURCES) $(mtp_trexist_SOURCES)
 ETAGS = etags
@@ -294,9 +280,6 @@
 mtp_files_SOURCES = files.c util.c util.h common.h
 mtp_folders_SOURCES = folders.c util.c util.h common.h
 mtp_trexist_SOURCES = trexist.c util.c util.h common.h
-mtp_playlists_SOURCES = playlists.c util.c util.h common.h
-mtp_getplaylist_SOURCES = getplaylist.c util.c util.h common.h
-mtp_newplaylist_SOURCES = newplaylist.c util.c util.h common.h
 mtp_format_SOURCES = format.c util.c util.h common.h
 mtp_albumart_SOURCES = albumart.c util.c util.h common.h
 mtp_albums_SOURCES = albums.c util.c util.h common.h
@@ -411,15 +394,6 @@
 mtp-format$(EXEEXT): $(mtp_format_OBJECTS) $(mtp_format_DEPENDENCIES) $(EXTRA_mtp_format_DEPENDENCIES) 
 	@rm -f mtp-format$(EXEEXT)
 	$(LINK) $(mtp_format_OBJECTS) $(mtp_format_LDADD) $(LIBS)
-mtp-getplaylist$(EXEEXT): $(mtp_getplaylist_OBJECTS) $(mtp_getplaylist_DEPENDENCIES) $(EXTRA_mtp_getplaylist_DEPENDENCIES) 
-	@rm -f mtp-getplaylist$(EXEEXT)
-	$(LINK) $(mtp_getplaylist_OBJECTS) $(mtp_getplaylist_LDADD) $(LIBS)
-mtp-newplaylist$(EXEEXT): $(mtp_newplaylist_OBJECTS) $(mtp_newplaylist_DEPENDENCIES) $(EXTRA_mtp_newplaylist_DEPENDENCIES) 
-	@rm -f mtp-newplaylist$(EXEEXT)
-	$(LINK) $(mtp_newplaylist_OBJECTS) $(mtp_newplaylist_LDADD) $(LIBS)
-mtp-playlists$(EXEEXT): $(mtp_playlists_OBJECTS) $(mtp_playlists_DEPENDENCIES) $(EXTRA_mtp_playlists_DEPENDENCIES) 
-	@rm -f mtp-playlists$(EXEEXT)
-	$(LINK) $(mtp_playlists_OBJECTS) $(mtp_playlists_LDADD) $(LIBS)
 mtp-reset$(EXEEXT): $(mtp_reset_OBJECTS) $(mtp_reset_DEPENDENCIES) $(EXTRA_mtp_reset_DEPENDENCIES) 
 	@rm -f mtp-reset$(EXEEXT)
 	$(LINK) $(mtp_reset_OBJECTS) $(mtp_reset_LDADD) $(LIBS)
@@ -450,11 +424,8 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/folders.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/format.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getfile.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getplaylist.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/newfolder.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/newplaylist.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pathutils.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/playlists.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/reset.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sendfile.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sendtr.Po@am__quote@
Index: src/libmtp.c
===================================================================
--- src/libmtp.c	(revision 148515)
+++ src/libmtp.c	(working copy)
@@ -40,7 +40,9 @@
 #include "ptp.h"
 #include "libusb-glue.h"
 #include "device-flags.h"
+#if 0
 #include "playlist-spl.h"
+#endif
 #include "util.h"
 
 #include <stdlib.h>
@@ -6764,6 +6766,7 @@
   return ret;
 }
 
+#if 0
 /**
  * This function renames a single playlist object file holder.
  * This simply means that the <code>PTP_OPC_ObjectFileName</code>
@@ -6800,6 +6803,7 @@
   playlist->name = strdup(newname);
   return ret;
 }
+#endif
 
 /**
  * This function renames a single album.
@@ -7217,6 +7221,7 @@
   return new_id;
 }
 
+#if 0
 /**
  * This creates a new playlist metadata structure and allocates memory
  * for it. Notice that if you add strings to this structure they
@@ -7413,6 +7418,7 @@
 
   return pl;
 }
+#endif
 
 /**
  * This function creates a new abstract list such as a playlist
@@ -7967,6 +7973,7 @@
 }
 
 
+#if 0
 /**
  * This routine creates a new playlist based on the metadata
  * supplied. If the <code>tracks</code> field of the metadata
@@ -8065,6 +8072,7 @@
 			      metadata->tracks,
 			      metadata->no_tracks);
 }
+#endif
 
 /**
  * This creates a new album metadata structure and allocates memory
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 148515)
+++ src/Makefile.am	(working copy)
@@ -1,9 +1,9 @@
 lib_LTLIBRARIES = libmtp.la
 
 libmtp_la_CFLAGS = @LIBUSB_CFLAGS@
-libmtp_la_SOURCES = libmtp.c unicode.c unicode.h util.c util.h playlist-spl.c \
+libmtp_la_SOURCES = libmtp.c unicode.c unicode.h util.c util.h \
 	gphoto2-endian.h _stdint.h ptp.c ptp.h libusb-glue.h \
-	music-players.h device-flags.h playlist-spl.h mtpz.h
+	music-players.h device-flags.h mtpz.h
 
 if LIBUSB1_COMPILE
 libmtp_la_SOURCES += libusb1-glue.c
Index: src/libmtp.h.in
===================================================================
--- src/libmtp.h.in	(revision 148515)
+++ src/libmtp.h.in	(working copy)
@@ -966,7 +966,7 @@
 int LIBMTP_Set_Folder_Name(LIBMTP_mtpdevice_t *, LIBMTP_folder_t *, const char *);
 /** @} */
 
-
+#if 0
 /**
  * @}
  * @defgroup playlists The audio/video playlist management API.
@@ -979,6 +979,7 @@
 int LIBMTP_Create_New_Playlist(LIBMTP_mtpdevice_t *, LIBMTP_playlist_t * const);
 int LIBMTP_Update_Playlist(LIBMTP_mtpdevice_t *, LIBMTP_playlist_t * const);
 int LIBMTP_Set_Playlist_Name(LIBMTP_mtpdevice_t *, LIBMTP_playlist_t *, const char *);
+#endif
 
 /**
  * @}
Index: src/Makefile.in
===================================================================
--- src/Makefile.in	(revision 148515)
+++ src/Makefile.in	(working copy)
@@ -85,16 +85,16 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 am__libmtp_la_SOURCES_DIST = libmtp.c unicode.c unicode.h util.c \
-	util.h playlist-spl.c gphoto2-endian.h _stdint.h ptp.c ptp.h \
-	libusb-glue.h music-players.h device-flags.h playlist-spl.h \
+	util.h gphoto2-endian.h _stdint.h ptp.c ptp.h \
+	libusb-glue.h music-players.h device-flags.h \
 	mtpz.h mtpz.c libusb1-glue.c libusb-glue.c libopenusb1-glue.c
 @MTPZ_COMPILE_TRUE@am__objects_1 = libmtp_la-mtpz.lo
 @LIBUSB1_COMPILE_TRUE@am__objects_2 = libmtp_la-libusb1-glue.lo
 @LIBUSB0_COMPILE_TRUE@am__objects_3 = libmtp_la-libusb-glue.lo
 @LIBOPENUSB_COMPILE_TRUE@am__objects_4 =  \
 @LIBOPENUSB_COMPILE_TRUE@	libmtp_la-libopenusb1-glue.lo
 am_libmtp_la_OBJECTS = libmtp_la-libmtp.lo libmtp_la-unicode.lo \
-	libmtp_la-util.lo libmtp_la-playlist-spl.lo libmtp_la-ptp.lo \
+	libmtp_la-util.lo libmtp_la-ptp.lo \
 	$(am__objects_1) $(am__objects_2) $(am__objects_3) \
 	$(am__objects_4)
 libmtp_la_OBJECTS = $(am_libmtp_la_OBJECTS)
@@ -252,8 +252,8 @@
 lib_LTLIBRARIES = libmtp.la
 libmtp_la_CFLAGS = @LIBUSB_CFLAGS@
 libmtp_la_SOURCES = libmtp.c unicode.c unicode.h util.c util.h \
-	playlist-spl.c gphoto2-endian.h _stdint.h ptp.c ptp.h \
-	libusb-glue.h music-players.h device-flags.h playlist-spl.h \
+	gphoto2-endian.h _stdint.h ptp.c ptp.h \
+	libusb-glue.h music-players.h device-flags.h \
 	mtpz.h $(am__append_1) $(am__append_2) $(am__append_3) \
 	$(am__append_4)
 include_HEADERS = libmtp.h
@@ -373,7 +373,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-libusb-glue.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-libusb1-glue.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-mtpz.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-playlist-spl.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-ptp.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-unicode.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libmtp_la-util.Plo@am__quote@
@@ -419,13 +418,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(LIBTOOL)  --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libmtp_la_CFLAGS) $(CFLAGS) -c -o libmtp_la-util.lo `test -f 'util.c' || echo '$(srcdir)/'`util.c
 
-libmtp_la-playlist-spl.lo: playlist-spl.c
-@am__fastdepCC_TRUE@	$(LIBTOOL)  --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libmtp_la_CFLAGS) $(CFLAGS) -MT libmtp_la-playlist-spl.lo -MD -MP -MF $(DEPDIR)/libmtp_la-playlist-spl.Tpo -c -o libmtp_la-playlist-spl.lo `test -f 'playlist-spl.c' || echo '$(srcdir)/'`playlist-spl.c
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/libmtp_la-playlist-spl.Tpo $(DEPDIR)/libmtp_la-playlist-spl.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='playlist-spl.c' object='libmtp_la-playlist-spl.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(LIBTOOL)  --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libmtp_la_CFLAGS) $(CFLAGS) -c -o libmtp_la-playlist-spl.lo `test -f 'playlist-spl.c' || echo '$(srcdir)/'`playlist-spl.c
-
 libmtp_la-ptp.lo: ptp.c
 @am__fastdepCC_TRUE@	$(LIBTOOL)  --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libmtp_la_CFLAGS) $(CFLAGS) -MT libmtp_la-ptp.lo -MD -MP -MF $(DEPDIR)/libmtp_la-ptp.Tpo -c -o libmtp_la-ptp.lo `test -f 'ptp.c' || echo '$(srcdir)/'`ptp.c
 @am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/libmtp_la-ptp.Tpo $(DEPDIR)/libmtp_la-ptp.Plo
