Index: src/libmtp.c
===================================================================
--- src/libmtp.c	(revision 148646)
+++ src/libmtp.c	(working copy)
@@ -1893,15 +1893,15 @@
   /* Check: if this is a PTP device, is it really tagged as MTP? */
   if (current_params->deviceinfo.VendorExtensionID != 0x00000006) {
     LIBMTP_ERROR("LIBMTP WARNING: no MTP vendor extension on device "
-		 "%d on bus %d",
+		 "%d on bus %d\n",
 		 rawdevice->devnum, rawdevice->bus_location);
-    LIBMTP_ERROR("LIBMTP WARNING: VendorExtensionID: %08x",
+    LIBMTP_ERROR("LIBMTP WARNING: VendorExtensionID: %08x\n",
 		 current_params->deviceinfo.VendorExtensionID);
-    LIBMTP_ERROR("LIBMTP WARNING: VendorExtensionDesc: %s",
+    LIBMTP_ERROR("LIBMTP WARNING: VendorExtensionDesc: %s\n",
 		 current_params->deviceinfo.VendorExtensionDesc);
     LIBMTP_ERROR("LIBMTP WARNING: this typically means the device is PTP "
 		 "(i.e. a camera) but not an MTP device at all. "
-		 "Trying to continue anyway.");
+		 "Trying to continue anyway.\n");
   }
 
   parse_extension_descriptor(mtp_device,
