Index: nspr/pr/src/misc/prdtoa.c
===================================================================
--- nspr/pr/src/misc/prdtoa.c	(revision 228201)
+++ nspr/pr/src/misc/prdtoa.c	(working copy)
@@ -13,6 +13,12 @@
 #pragma GCC diagnostic ignored "-Wparentheses"
 #endif
 
+#if _MSC_VER == 1800
+// TODO(scottmg): Crashes VS2013 RC. Upstream bug filed and should be fixed
+// in the next release. http://crbug.com/288948.
+#pragma optimize("", off)
+#endif
+
 #include "primpl.h"
 #include "prbit.h"
 
